### PR TITLE
feat(queue): enhance getJobScheduler method to include template information

### PR DESCRIPTION
**Pull request by roggervalf**
ref https://github.com/taskforcesh/bullmq/issues/2875

### CODE REVIEW

File: src/classes/job-scheduler.ts [hunk 1]
```diff
@@ -241,18 +248,28 @@ export class JobScheduler extends QueueBase {
...
-        tz: jobData.tz || null,
-        pattern: jobData.pattern || null,
-        every: jobData.every || null,
+        name: schedulerAttributes.name,
+        endDate: parseInt(schedulerAttributes.endDate) || null,
+        tz: schedulerAttributes.tz || null,
+        pattern: schedulerAttributes.pattern || null,
+        every: schedulerAttributes.every || null,
+        ...(schedulerAttributes.data
+          ? {
```
Comment (manast) [inline src/classes/job-scheduler.ts:266]: If you are checking if data is defined here, why having a default in the next line?:
```data: JSON.parse(schedulerAttributes.data || '{}')```
Also, could it potentially be the case that you have an empty data but options?

File: tests/test_job_scheduler.ts [hunk 2]
```diff
@@ -360,6 +360,12 @@ describe('Job Scheduler', function () {
       tz: null,
       pattern: '*/2 * * * * *',
       every: null,
+      template: {
```
Comment (manast) [inline tests/test_job_scheduler.ts:363]: Maybe add cases where data / opts are undefined?

File: src/classes/job-scheduler.ts (hunk 1 above)
Comment (roggervalf) [reply src/classes/job-scheduler.ts:252]: yeah I refactored a little bit to not save this data if it's empty, same for template opts

File: tests/test_job_scheduler.ts (hunk 2 above)
Comment (roggervalf) [reply tests/test_job_scheduler.ts:363]: addressed as well

**Review by manast — APPROVED** [review]
LGTM

**Pull request merged**